solution/index.js

```diff
@@ -0,0 +1,349 @@
+        newLocalStorageArray.unshift(targetElement.innerText)
+        localStorage.tasks = JSON.stringify(localStorageObject)
+        window.onkeydown = null;
+    };
+}
+
+//Deleting of all <li> elements and generates anew
+function refreshTaskSection(){
+    const allListsOfTasks = document.querySelectorAll("ul")
+    for(let list of allListsOfTasks){
+        while (list.firstChild) {
+            list.removeChild(list.firstChild);
+        }
+    }
+    generationTasklistFromLocalStorage(localStorageObject)
+    generationImportantTasksFromLocalStorage(importantTasksArray)
+}
+
+//Fucntion of search input that do refresh of all tasks and sort to you all tasks that include text from input area
+function searchTasks(){
+    const input = document.getElementById("search")
+    const valueInInput = input.value.toLowerCase()
+    refreshTaskSection();
+    const allTasks = document.querySelectorAll(".task")
+    for(let task of allTasks){
+        const taskValue = task.innerText.toLowerCase()
+        if(!taskValue.includes(valueInInput)){
+            task.parentElement.removeChild(task)
+        }
+    }
+}
+
+//Fucntion of save tasks in API Storage
+async function saveToStorage(){
+    displayLoading();
+
+    const response = await fetch("https://json-bins.herokuapp.com/bin/614ee0bfc092dca86cb87689", { //send request to API
+        method: "PUT" ,
+        headers: {
+            'Accept': "application/json" ,
+            'Content-Type': "application/json",
+        },
+        body: JSON.stringify({"tasks": localStorageObject})
+    })
+
+    hideLoading();
+
+    if(!response.ok){ // error part: behavior of all elements in error situation anf throw ERROR with current status
+        alert("ERROR "+response.status+". "+response.statusText)//displaying in result element at error message
+        throw `ERROR ${response.status}`
+    }
+    
+}
+
+////Fucntion of loading tasks from API Storage if in localStorage we have other tasks function refreshes localStorage information to API information 
+async function loadFromStorage(){
+    displayLoading();
+    const response =await fetch("https://json-bins.herokuapp.com/bin/614ee0bfc092dca86cb87689")
+    if(!response.ok){ // error part: behavior of all elements in error situation anf throw ERROR with current status
+        alert("ERROR "+response.status+". "+response.statusText)//displaying in result element at error message
+        throw `ERROR ${response.status}`
+    }
+    hideLoading();
+    const tasksOBjectFromStorage=(await response.json()).tasks
+    localStorageObject=tasksOBjectFromStorage;
+    localStorage.tasks=JSON.stringify(localStorageObject);
+    refreshTaskSection();
+    
+}
+
+//Starting drag event
+function dragEvent (event){
+    draggingElement = event.target.closest("li")
+    event.dataTransfer.setData("text", event.target.closest("ul").className);
+
+    const sections = document.querySelectorAll("section") //Adding event listener to all sections
```

Comment:
There is no reason to add and remove events from the section by clicking on the li. There are better logical ways to perform the requested action